go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -166,6 +166,9 @@ func newDefaultNetworkControllerCommon(cnci *CommonNetworkControllerInfo,
 	if config.OVNKubernetesFeature.EnableInterconnect {
 		zoneICHandler = zoneic.NewZoneInterconnectHandler(&util.DefaultNetInfo{}, cnci.nbClient, cnci.sbClient)
 		zoneChassisHandler = zoneic.NewZoneChassisHandler(cnci.sbClient)
+		if err := zoneICHandler.Init(cnci.nodeName, cnci.kube, ctx); err != nil {
+			return nil, err
+		}
```

Comment:
Wouldn't this fit better in the `Start` methods of the controllers, where you already have a context users are already expecting for it that it might take some time,rather than the constructor?

Reply:
yeah I'll move it